libs/execution_context/ebpf_core.c

```diff
@@ -1734,51 +1734,70 @@ typedef struct _ebpf_protocol_handler
     } dispatch;
     size_t minimum_request_size;
     size_t minimum_reply_size;
+    union
+    {
+        uint64_t value;
+        struct
+        {
+            uint64_t used_by_native : 1;
+            uint64_t used_by_jit : 1;
+            uint64_t used_by_interpret : 1;
+        } bits;
+    } flags;
+
 } const ebpf_protocol_handler_t;
 
-#define DECLARE_PROTOCOL_HANDLER_FIXED_REQUEST_NO_REPLY(OPERATION)                    \
+#define PROTOCOL_NATIVE_MODE 1
+#define PROTOCOL_JIT_MODE 2
+#define PROTOCOL_INTERPRET_MODE 4
+#define PROTOCOL_JIT_OR_INTERPRET_MODE 6
+#define PROTOCOL_ALL_MODES 7
+
+#define DECLARE_PROTOCOL_HANDLER_FIXED_REQUEST_NO_REPLY(OPERATION, FLAGS)             \
```

Comment:
suggestion: modify the macros such that ALL_MODES is the default value of the flags.